workers/deployment_manager/DeploymentManager/Worker/DeploymentManager.cs

```diff
@@ -0,0 +1,193 @@
+
+namespace DeploymentManager
+{
+    internal class DeploymentManager
+    {
+        public static string WorkerType = "DeploymentManager";
+
+        private string projectName;
+        private string assemblyName;
+        private string deploymentNamePrefix;
+        private string devAuthToken;
+        private int numberOfDeployments;
+        private string clientType;
+        private int maxPlayerCount;
+        private int playerCount;
+
+        private Connection metaConnection;
+
+        public DeploymentManager(string receptionistHost, ushort receptionistPort, string workerId, string projectName, string clientType,
+            string assemblyName, string deploymentNamePrefix, int numberOfDeployments, int maxPlayerCount)
+        {
+            this.projectName = projectName;
+            this.assemblyName = assemblyName;
+            this.deploymentNamePrefix = deploymentNamePrefix;
+            this.numberOfDeployments = numberOfDeployments;
+            this.clientType = clientType;
+            this.maxPlayerCount = maxPlayerCount;
+
+            var serviceAccountToken = GetFileContent("ServiceAccountToken.txt");
+            DeploymentModifier.Init(serviceAccountToken);
+            WorkerAuthenticator.Init(serviceAccountToken);
+
+            // Connect to meta deployment
+            var connectionParameters = new ConnectionParameters
+            {
+                WorkerType = WorkerType,
+            };
+
+            var connector = new Connector(WorkerType, string.Empty);
+            metaConnection = connector.TryToConnect(connectionParameters, receptionistHost, receptionistPort, workerId);
+
+            devAuthToken = WorkerAuthenticator.RetrieveDevelopmentAuthenticationToken(projectName);
+        }
+
+        private string GetFileContent(string fileName)
+        {
+            string filePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, fileName);
+            if (File.Exists(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, fileName)))
+            {
+                Log.Print(LogLevel.Warn, $"Was not able to find file in path {filePath}");
+            }
+
+            return File.ReadAllText(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, fileName));
+        }
+
+        public void ObserveDeployments()
+        {
+            // spin up the other deployments
+            var snapshotPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "default.snapshot");
+            var launchConfig = new LaunchConfig
+            {
+                ConfigJson = GetFileContent("launch_config.json"),
+            };
+
+            Log.Print(LogLevel.Info, $"Number of deployments {numberOfDeployments}");
+            for (var i = 0; i < numberOfDeployments; i++)
+            {
+                var deploymentName = $"{deploymentNamePrefix}_{i}";
+                Task.Run(() => StartDeployment(deploymentName, snapshotPath, launchConfig));
```

Comment:
This is an odd use of the `Task.Run` API, which isn't meant for long-running tasks like this. You may be better off using the explicit `Thread` API to start a heavyweight thread up.